transactions/models.py

```diff
@@ -0,0 +1,77 @@
+from django.db import models
+
+# Create your models here.
+from blocks.models import Block
+
+
+class Transaction(models.Model):
+    '''
+    This represents a transaction(exchange of coins) on the blockchain
+    '''
+
+    # hash of the transaction uniquely identifies it
+    hash = models.CharField(max_length=40, editable=False, blank=True, null=True)
+
+    # block this transaction is part of (if it has been mined that is)
+    block = models.ForeignKey(Block, blank=True, null=True, on_delete=models.CASCADE, related_name='transactions')
+
+    # number of inputs in the transaction
+    tx_ins_ct = models.PositiveIntegerField(blank=True, default=0)
+
+    # number of outputs in the transaction
+    tx_outs_ct = models.PositiveIntegerField(blank=True, default=0)
+
+    # track if the transaction has been mined
+    mined = models.BooleanField(default=False)
+
+    def __str__(self):
+        return f"tx hash {self.hash} in block {self.block.hash}"
+
+
+class AbstractTransactionComponent(models.Model):
+    '''
+    This is an abstract model specifying the common functionality
+    of transaction inputs and transaction outputs
+    '''
+
+    # hash of the transaction component
+    hash = models.CharField(max_length=40, editable=False, blank=True, null=True)
+
+    # value of the component i.e how much is it worth
+    value = models.PositiveIntegerField(default=0)
+
+    # who owns/owned this component
+    own_addr = models.CharField(max_length=100)
+
+    # in the transaction that generated this component, what was its index among all outputs
+    gen_tx_idx = models.PositiveIntegerField(blank=True, null=True)
+
+    class Meta:
+        abstract = True
+
+
+class TxIn(AbstractTransactionComponent):
```

Comment:
The model name should be descriptive and readable. TxIn does not give any context from its name.